### ISSUE TITLE
possible installation issue... 

### ISSUE THREAD

**Issue by kilgore5** [kingandpartners/slate#2]
John:squirrel:  11:24 AM
can you install this new version of Slate from the KP repo and let me know how your package-lock changes?

Jibin  12:40 PM
sorry just wanted to make sure i got these steps right before i bring up this issue
Before running each method i deleted node_modules and package-lock.json
Method 1: If I run npm i on my current node modules and then install those 2 slate modules seperately I get 0 issues after running npm run start
Method 2: If I run npm i with those 2 modules saved in package.json I get the following error
Failed to compile.

./scripts/layout/theme.js
Module build failed (from ../node_modules/@shopify/slate-tools/node_modules/babel-loader/lib/index.js):
Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
    at throwVersionError (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@babel/helper-plugin-utils/lib/index.js:78:11)
    at Object.assertVersion (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@babel/helper-plugin-utils/lib/index.js:28:5)
    at /home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@babel/plugin-syntax-dynamic-import/lib/index.js:11:7
    at /home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@babel/helper-plugin-utils/lib/index.js:22:12
    at Function.memoisePluginContainer (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-loader/lib/index.js:50:20)
    at Object.module.exports (/home/jibin/Desktop/projects/shopify-boilerplate/node_modules/@shopify/slate-tools/node_modules/babel-loader/lib/index.js:173:20)
 @ multi ../node_modules/@shopify/slate-tools/tools/webpack/hot-client.js ./scripts/layout/theme.js layout.theme[1]

John:squirrel:  12:46 PM
for method 2 you mean you just added the lines manually to package.json vs. method 1 running npm i [url] [url] ?

Jibin  12:47 PM
yes for method 2 its saved in package.json vs running the npm commands (edited) 

John:squirrel:  12:48 PM
ok, I don't think npm likes method 2 in general, esp. when trying to install 2 packages where one is a dependency of the other
12:49
npm dependency management is insane

Jibin  12:50 PM
so for this update the dev would have to run those extra steps in order to get the repo to work?

John:squirrel:  12:51 PM
well I think just using method 1 is the way to go
12:51
don't most people use that method anyway?

Jibin  12:51 PM
usually after slate is up and running
12:52
i'm just trying to reduce the amount of steps for setup (edited) 

John:squirrel:  12:53 PM
hmm... I'm not quite clear on the difference then
12:53
I would think you'd always need to restart slate after running npm i

Jibin  12:55 PM
yes, but i'm saying ideally all required packages would be set in package.json and the developer would just have to run npm i and they can get started with development
But its very little work to run 2 extra commands before starting so its not a huge deal

John:squirrel:  12:56 PM
right but once the packages are installed they don't have to install them again?

Jibin  12:59 PM
eh, i've had some rare occassions where i had to delete node_modules and reinstall

John:squirrel:  1:09 PM
ok, I see what you're saying now... had to read back to what you originally said
1:11
I think the real test is:
installing the 2 new packages via npm i [package] [package], which updates package.json and package-lock
then delete `node_module`s folder
then run plain npm i again
1:12
we should (almost) almost never have a case with a fresh install without both package-lock and node_modules

Jibin  1:15 PM
jibin@pop-os:~/Desktop/projects/shopify-boilerplate$ npm i
npm WARN deprecated @babel/polyfill@7.12.1: :rotating_light: This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated @babel/preset-es2015@7.0.0-beta.53: :wave: We've deprecated any official yearly presets in 6.x in favor or babel-preset-env. For 7.x it would be @babel/preset-env.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @shopify/slate-sync@1.0.1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of '@shopify/slate-tools'
npm ERR! notarget 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jibin/.npm/_logs/2021-04-08T17_14_49_289Z-debug.log
1:15
thats what i get after running that test

John:squirrel:  1:15 PM
yep, I get the same thing... so not great (edited) 

John:squirrel:  1:24 PM
if I run npm i --save-dev https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-sync-1.0.1.tgz https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-tools-1.0.1.tgz again though npm seems to fix itself
1:25
so we could add a note about this to the readme and/or add a "setup" command to package.json to make it more obvious what the intent is
1:26
alternately we could add the tarballs to the repo in node_vendor and reference them there... might be slightly less confusing

Jibin  1:58 PM
"scripts": {
    "setup": "npm i --save-dev https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-sync-1.0.1.tgz && npm i --save-dev https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-tools-1.0.1.tgz && npm i"
  },
(edited)
1:58
so like that?

John:squirrel:  2:00 PM
yeah, basically thought I'd do the 2 slate packages at the same time, which seems to cause npm less consternation due to the dependencies
2:00
"scripts": {
    "setup": "npm i --save-dev https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-sync-1.0.1.tgz https://github.com/kingandpartners/slate/releases/download/v1.0.1/shopify-slate-tools-1.0.1.tgz && npm i"
  },

Jibin  2:01 PM
do you see the same babel error that i'm seeing? (edited) 

John:squirrel:  2:02 PM
I did previously
2:02
but once I run the "setup" command it seems to be fine
2:02
what about you?

Jibin  2:03 PM
what did you start with? you had both node_modules and package-lock.json still intact? (edited) 

John:squirrel:  2:03 PM
no, I only deleted node_modules folder

Jibin  2:09 PM
hmm still seeing the issue

John:squirrel:  2:09 PM
hmm, ok... let me try again

John:squirrel:  3:05 PM
ok... confirmed
deleted node_modules folder.
package.json has the new slate packages, as does package-lock.json
if I run npm run setup then everything installs again for me and I can run npm run deploy correctly
3:06
but you're still getting No matching version found for @shopify/slate-sync@1.0.1. error?

Jibin  3:08 PM
no its the babel error when i run npm run start

John:squirrel:  3:08 PM
oh, right
3:09
I'm using "babel-preset-shopify": "^16.7.0",
3:09
you are probably using something different?

Jibin  3:10 PM
oh im using "@shopify/babel-preset": "23.4.0"
:+1:
1

John:squirrel:  3:16 PM
I swapped to @shopify/babel-preset  when I was working on IE fixes but those never got merged/deployed while I was out... I'll try in that branch and see what happens
3:17
on that note, would really like to have clean babel setup already for IE in our boilerplate
3:18
I'll get back to Slate stuff tonight
3:18
back to merit bundling for now

**Comment by urielking** [comment]
To fix the babel dependency issue noted above - use npm version 8.3, and add the following to your project's `package.json`
```
"overrides": {
    "@shopify/themekit": "^1.1.7",
    "@shopify/slate-tools": {
      "babel-core": "^7.0.0-bridge.0"
    }
  },
```
